packages/vendure-plugin-customer-managed-groups/src/customer-managed-groups.service.ts

```diff
@@ -364,4 +371,140 @@ export class CustomerManagedGroupsService {
       isGroupAdministrator,
     };
   }
+
+  async updateGroupMember(
+    ctx: RequestContext,
+    input: UpdateCustomerManagedGroupMemberInput
+  ): Promise<CustomerManagedGroup> {
+    if (
+      !input.title &&
+      !input.firstName &&
+      !input.lastName &&
+      !input.emailAddress
+    ) {
+      throw new UserInputError(`Make sure to include fields to be updated`);
+    }
+    const myGroup = await this.myCustomerManagedGroup(ctx);
+    if (myGroup) {
+      const groupAdmin = myGroup.customers.find((c) => c.isGroupAdministrator);
+      if (!groupAdmin) {
+        throw new UserInputError('group has no admin');
+      }
+      const groupAdminCustomer = await this.customerService.findOne(
+        ctx,
+        groupAdmin?.customerId,
+        ['user']
+      );
+      const member = myGroup.customers.find(
+        (customer) => customer.customerId === input.customerId
+      );
+      if (member) {
+        const userRepo = getRepository(User);
+        const customer = await this.customerService.findOne(
+          ctx,
+          member.customerId,
+          ['user']
+        );
+        if (!customer || !customer.user) {
+          throw new UserInputError(
+            `No customer with id ${input.customerId} exists`
+          );
+        }
+        if (
+          groupAdminCustomer?.user?.id != ctx.activeUserId &&
+          customer.user.id != ctx.activeUserId
+        ) {
+          throw new UserInputError(
+            `You are not allowed to update other member's details`
+          );
+        }
+        if (input.emailAddress) {
+          if (
+            await userRepo.count({
+              where: { identifier: input.emailAddress },
+            })
+          ) {
+            throw new UserInputError('User with this email already exists');
+          }
+        }
+        const updateUserData = {
+          id: customer.id,
+          ...(input.title ? { title: input.title } : []),
+          ...(input.firstName ? { firstName: input.firstName } : []),
+          ...(input.lastName ? { lastName: input.lastName } : []),
+          ...(input.emailAddress ? { emailAddress: input.emailAddress } : []),
+        };
+        await this.customerService.update(ctx, updateUserData);
+        const newGroupData = await this.myCustomerManagedGroup(ctx);
+        return newGroupData!;
+      }
+      throw new UserInputError(
+        `No customer with id ${input.customerId} exists in '${myGroup.name}' customer managed group`
+      );
+    }
+    throw new UserInputError(
+      `No customer managed group exists for the authenticated customer`
+    );
+  }
```

Comment:
```suggestion
  async updateGroupMember(
    ctx: RequestContext,
    input: UpdateCustomerManagedGroupMemberInput
  ): Promise<CustomerManagedGroup> {
    if (
      !input.title &&
      !input.firstName &&
      !input.lastName &&
      !input.emailAddress
    ) {
      throw new UserInputError(`Make sure to include fields to be updated`);
    }
    if (!ctx.activeUserId) {
      throw new ForbiddenError();
    }
    const myGroup = await this.myCustomerManagedGroup(ctx);
    if (!myGroup) {
      throw new UserInputError(
        `No customer managed group exists for the authenticated customer`
      );
    }
    const groupAdmin = myGroup.customers.find((c) => c.isGroupAdministrator);
    if (!groupAdmin) {
      throw new UserInputError('Group has no admin');
    }
    if (!this.isAdministratorOfGroup(ctx.activeUserId, myGroup) && input.customerId.user.id !== ctx.activeUserId) {
      throw new UserInputError(
        `You are not allowed to update other member's details`
      );
    }
    const member = myGroup.customers.find(
      (customer) => customer.customerId === input.customerId
    );
    if (!member) {
      throw new UserInputError(`Your group doesn't have a member with customerId ${input.customerId}`);
    }
    const customer = await this.customerService.findOne(
      ctx,
      member.customerId,
      ['user']
    );
    if (!customer || !customer.user) {
      throw new UserInputError(
        `No customer with id ${input.customerId} exists`
      );
    }
    const userRepo = getRepository(User);
    if (input.emailAddress && await userRepo.count({
      where: { identifier: input.emailAddress },
    })) {
      throw new UserInputError('User with this email already exists');
    }
    const updateUserData = {
      id: customer.id,
      ...(input.title ? { title: input.title } : []),
      ...(input.firstName ? { firstName: input.firstName } : []),
      ...(input.lastName ? { lastName: input.lastName } : []),
      ...(input.emailAddress ? { emailAddress: input.emailAddress } : []),
    };
    await this.customerService.update(ctx, updateUserData);
    const newGroupData = await this.myCustomerManagedGroup(ctx);
    if (!newGroupData) {
      throw Error(`Group with id ${myGroup.id} not found`); // Should never happen
    }
    return newGroupData!;
  }
```

Hi Surafel. I think this piece of code could be simplified a bit more. Here is a suggestion, (it's up to you if you accept the suggestion in your PR). Generally speaking, you should have only 1 or 2 levels of nested if/else statements. If you have more, then in most cases this can be considered a code-smell. There are of course exceptions.